### PR TITLE
[server] Use redlock for webhook gc WEB-224

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-server-0a41d11a34</li>
	<li><b>🔗 URL</b> - <a href="https://mp-server-0a41d11a34.preview.gitpod-dev.com/workspaces" target="_blank">mp-server-0a41d11a34.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/server/src/projects/webhook-event-garbage-collector.ts [hunk 1]
```diff
@@ -8,31 +8,45 @@ import { injectable, inject } from "inversify";
...
-    public async start(intervalSeconds?: number): Promise<Disposable> {
-        const intervalSecs = intervalSeconds || WebhookEventGarbageCollector.GC_CYCLE_INTERVAL_SECONDS;
+    public async start(): Promise<Disposable> {
+        const intervalMs = WebhookEventGarbageCollector.GC_CYCLE_INTERVAL_SECONDS * 1000;
         return repeat(async () => {
             try {
-                if (await this.leaderQuorum.areWeLeader()) {
-                    await this.collectObsoleteWebhookEvents();
-                }
+                await this.mutex.using(["workspace-gc"], intervalMs, async (signal) => {
```
Comment (geropl) [inline components/server/src/projects/webhook-event-garbage-collector.ts:30]: After reading this code four times now: Would be awesome to have a common interface between those GCs, and register them in a central place.

But out of scope here, now, I guess.

Comment (easyCZ) [reply components/server/src/projects/webhook-event-garbage-collector.ts:30]: Yep, already planning to consolidate jobs into a singular place, just wanted to do these in place before we do the bigger move

**Review by geropl — APPROVED** [review]
:heavy_check_mark:

**Pull request merged**